File: app/src/main/java/com/softsquared/android/mask_app/src/splash/SplashActivity.java
package com.softsquared.android.mask_app.src.splash;

import android.content.Intent;
import android.os.Bundle;

import com.softsquared.android.mask_app.R;
import com.softsquared.android.mask_app.src.BaseActivity;
import com.softsquared.android.mask_app.src.splash.interfaces.SplashActivityView;


public class SplashActivity extends BaseActivity implements SplashActivityView {
    private static final String TAG = "SPLASH ACT://";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        Thread splashThread;

        splashThread = new Thread() {
            @Override
            public void run() {
                try {
                    synchronized (this) {
                        // Wait given period of time or exit on touch
                        wait(1000);

                        Intent mainIntent = new Intent(SplashActivity.this, MainActivity.class);
                        startActivity(mainIntent);
                        finish();
                    }
                } catch (InterruptedException ignored) {

                }
            }
        };
        splashThread.start();
    }


    @Override
    public void validateFailure(String message) {

    }

    @Override
    public void validateSuccess(String message) {

    }
}

File: app/src/main/java/com/minux/mask_alarmi/presentation/splash/SplashActivity.kt
package com.minux.mask_alarmi.presentation.splash

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.minux.mask_alarmi.presentation.main.MainActivity
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class SplashActivity: AppCompatActivity(){
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val intent = Intent(this, MainActivity::class.java).apply {
            flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
        }
        startActivity(intent)
        finish()
    }
}